### ISSUE TITLE
Ingest Andersen Lab metadata and sequences

### ISSUE THREAD

**Issue by joverlee521** [nextstrain/avian-flu#26]
Jotting down the outline of ingesting the metadata and sequences from https://github.com/andersen-lab/avian-influenza

1. Fetch tarball of https://github.com/andersen-lab/avian-influenza
2. Concatenate FASTA files of same segment
3. Cut FASTA headers down to just the SRA run accession
4. Join segment FASTAs with metadata using `augur curate`
5. Curate metadata - matching [Nextstrain columns](https://github.com/moncla-lab/avian-flu-USDA-cattle/blob/main/metadata-with-clade-nextstrain-format.tsv) -> [Andersen Lab metadata](https://github.com/andersen-lab/avian-influenza/blob/master/metadata/PRJNA1102327_metadata.csv)

|Nextstrain       |Andersen Lab                                               |
|-----------------|-----------------------------------------------------------|
|strain           |A/<Host\>/USA/<isolate\>/<year\> \(<year\> parsed from <Date\>\)|
|virus            |avian\_flu                                                 |
|isolate\_id      |<Run\> \(SRA accession\)                                   |
|date             |<Date\> \(any missing date or ? marked date 2024\-XX\-XX\) |
|region           |North America                                              |
|country          |USA                                                        |
|division         |<US State\>                                                |
|location         |<US State\>                                                |
|host             |<Host\> (binned into Avian, Cattle, or Nonhuman Mammal) |
|domestic\_status |?                                                          |
|subtype|h5n1                                                                    |
|originating\_lab |<Center Name\>                                             |
|submitting\_lab  |<Center Name\>                                             |
|authors          |?                                                          |
|PMID             |?                                                          |
|gisaid\_clade    |?                                                          |
|h5\_clade        |?                                                          |

6. Output metadata.tsv and sequences.fasta per segment
7. Run metadata files through [add_segment_counts.py](https://github.com/nextstrain/avian-flu/blob/master/scripts/add_segment_counts.py) to match fauna downloaded data
8. Upload output files to S3
9. Concatenate fauna and Andersen lab files
10. Dedup based on `strain` name
11. Upload merged data to S3

**Comment by joverlee521** [comment]
WIP pushed up to https://github.com/nextstrain/avian-flu/tree/ingest-andersen-lab

Not hooked up to upload to S3 or merge/dedup with fauna data yet, but the ingest from the Andersen Lab's repo can be run with 

```
nextstrain build ingest merge_andersen_segment_metadata 
```

This will output the results to `ingest/results/andersen-lab/`.

**Comment by joverlee521** [comment]
I tried to ensure that the metadata output from this workflow exactly matches @lmoncla 's [metadata](https://github.com/moncla-lab/avian-flu-USDA-cattle/blob/main/metadata-with-clade-nextstrain-format.tsv). 

Currently there's just one difference in a strain name
```
A/Goose/USA/23-038138-001-original/2024->A/Goose/USA/23-038138-001-original/2023
```

**Comment by jameshadfield** [comment]
Nice work @joverlee521 - I ran this today with the following patch to simplify metadata merging and everything looks good

```diff
diff --git a/ingest/scripts/curate_andersen_lab_data.py b/ingest/scripts/curate_andersen_lab_data.py
index 4ded49e..a7b62d6 100644
--- a/ingest/scripts/curate_andersen_lab_data.py
+++ b/ingest/scripts/curate_andersen_lab_data.py
@@ -22,6 +22,7 @@ NEXTSTRAIN_RECORD = {
     'location': '?',
     'host': '?',
     'domestic_status': '?',
+    'subtype': 'h5n1',
     'originating_lab': '?',
     'submitting_lab': '?',
     'authors': '?',
```

**Comment by joverlee521** [comment]
Thanks for catching that @jameshadfield , not sure how I missed the `subtype` field!

**Comment by joverlee521** [comment]
Based on internal discussion ([Slack](https://bedfordlab.slack.com/archives/CD84ELG0N/p1715019891716949?thread_ts=1714683613.366519&cid=CD84ELG0N)), holding off on uploading Andersen lab data to S3 and merging with fauna data since the current sequences have become available via GISAID. 

Will make a PR to add the Andersen lab workflow to `ingest/` in case we need to ingest any new data in the future.

**Comment by joverlee521** [comment]
Closed by #28 - can revisit uploads to S3 and merging with fauna in a future issue.
